lib/game.rb

```diff
@@ -0,0 +1,79 @@
+require 'pry'
+class Game
+
+attr_accessor :board, :player_1, :player_2
+
+WIN_COMBINATIONS = [
+   [0,1,2],
+   [3,4,5],
+   [6,7,8],
+   [2,4,6],
+   [0,4,8],
+   [0,3,6],
+   [1,4,7],
+   [2,5,8]
+]
+
+
+def initialize(player_1 = Player::Human.new("X"), player_2 = Player::Human.new("O"), board = Board.new)
+@player_1 = player_1
+@player_2 = player_2
+@board = board
+end
+
+
+def current_player
+  turns = @board.cells.select {|piece| piece == "X" || piece == "O"}
+  if turns.count % 2 == 0
+  player_1
+else
+   player_2
+end
+end
+
+def over?
+ draw? || won?
+end
+
+def won?
+   WIN_COMBINATIONS.detect do |combination|
+     @board.cells[combination[0]] == @board.cells[combination[1]] && @board.cells[combination[1]]== @board.cells[combination[2]] && @board.cells[combination[0]] != " "
+end
+end
+
+def draw?
+ @board.full? && !won?
+end
+
+def winner
+  @board.cells[won?[0]] if won?
```

Comment:
Instead of calling `Game#won?` twice, consider using an if statement assignment. For example:

```ruby
if won = won?
  board.cells[won.first]
end
```

If `won?` returns true, then and only then will the local variable `won` be assigned, otherwise, the method will just abort with a false return value. But if `won` is assigned, we can use that variable instead of calling `won?` again within the if statement.